app/src/main/java/com/android/streetworkapp/ui/profile/Profile.kt

```diff
@@ -65,12 +75,27 @@ fun ProfileScreen(
               horizontalArrangement = Arrangement.spacedBy(10.dp, Alignment.CenterHorizontally),
               verticalAlignment = Alignment.Top) {
 
-                // profile placeholder
-                Image(
-                    painter = painterResource(id = R.drawable.profile),
-                    contentDescription = "profile picture",
-                    modifier = Modifier.size(200.dp).testTag("profilePicture"),
-                )
+                // fetch profile picture from firebase
+                val auth = Firebase.auth
+                val user = auth.currentUser
```

Comment:
From a more general point of view, this UI file will fetch the current user, and that user will be used across all fields in the UI. Wouldn’t it be better to fetch the current user outside of the `Box` element, rather than within this very specific part of the file?

Reply:
fixed

Reply:
I was more thinking of first fetching the users and then fetching the photo specifically. The user will be used in several parts of the file so it would be nice to have a global `user` variable that we can reuse to access `username`, `score`, etc.

Reply:
I dont think we need the firebase user, it is only used to fetch the gmail profile picture.
The nb of point and other feature are inside the user MVVM

Reply:
Indeed, I hadn't thought of that! In this case it's all good.